runtime/vm/JFRChunkWriter.cpp

```diff
@@ -109,7 +109,7 @@ VM_JFRChunkWriter::writeJFRMetadata()
 	}
 
 	/* add metadata size */
-	_bufferWriter->writeLEB128PaddedU32(dataStart, _bufferWriter->getCursor() - dataStart);
+	_bufferWriter->writeLEB128PaddedU32(dataStart, (U_32)(_bufferWriter->getCursor() - dataStart));
```

Comment:
Simply casting to `U_32` seems wrong to me.

This pattern occurs often enough that I suggest `VM_BufferWriter` should have a method to write the size:
```c
		_bufferWriter->writeSize(dataStart);
```
That method can use `this->getCursor()` to compute the size, validate that it fits in 28 bits and store it at `dataStart`.

BTW, the `U32` suffix might lead one to believe that sizes can be up to 32 bits wide, but that's not true: methods named with the `U32` suffix only support values up to 28 bits wide. Similar comments apply for the `U64` and `U72` suffixes. I think we should consider renaming those methods.

Reply:
To expand on the comment above: `VM_BufferWriter` should treat writing values that don't fit in the related format as a form of "overflow".

Reply:
I dont think we need both U_64 and U_72 so we can drop usages of U_64 and just use U_72 there.

Reply:
In the `writeLEB128PaddedU32` check that the input has the 4 MSB set to zero, if not `_overflow = true;`

Reply:
> I think `padded[Size]` makes it clear that the actual data is less than the memory consumed.

On the contrary, "padded" suggests to me that the output is _padded_ as necessary; the "U32" part says to me that the argument is 32 bits wide, which is not true.